mica/archive/aca_l0.py

```diff
@@ -769,14 +765,14 @@ def _fetch_individual_files(self, files):
 
     def _insert_files(self, files):
         count_inserted = 0
-        with Ska.DBI.DBI(**self.db) as db:
-            for i, f in enumerate(files):
-                arch_info = self._read_archfile(i, f, files)
-                if arch_info:
-                    self._move_archive_files([f])
+        for i, f in enumerate(files):
+            arch_info = self._read_archfile(i, f, files)
+            if arch_info:
+                self._move_archive_files([f])
+                with Ska.DBI.DBI(**self.db) as db:
                     db.insert(arch_info, 'archfiles')
-                    count_inserted += 1
-            db.commit()
+                    db.commit()
```

Comment:
It is less efficient here to commit after each insert, but the "move_archive_files" routine needs a lock on the db and it can't do that if there are uncommited inserts.